### PR TITLE
chore(TDI-39964): declare jars in parent for aligning versions

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-39964

**What is the new behavior?**
* Declared component's dependencies in `components-parent/pom.xml`
* Removed version in component's `pom.xml`

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1626/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1626/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1627/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1627/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1629/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1629/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1630/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1630/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1633/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1633/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1634/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1634/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1635/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1635/)

File: components-parent/pom.xml [hunk 1]
```diff
@@ -1,634 +1,1562 @@
...
+        <aws-java-sdk-core.version>${aws.version}</aws-java-sdk-core.version>
+        <aws-java-sdk-dynamodb.version>${aws.version}</aws-java-sdk-dynamodb.version>
+        <aws-java-sdk-kinesis.version>${aws.version}</aws-java-sdk-kinesis.version>
+        <aws-java-sdk-kms.version>${aws.version}</aws-java-sdk-kms.version>
+        <aws-java-sdk-s3.version>${aws.version}</aws-java-sdk-s3.version>
+        <aws-java-sdk-sts.version>${aws.version}</aws-java-sdk-sts.version>
+        <aws.version>1.11.125</aws.version>
+        <azure-keyvault-core.version>0.9.7</azure-keyvault-core.version>
+        <azure-storage.version>5.0.0</azure-storage.version>
+        <com.fasterxml.jackson.version>2.8.8</com.fasterxml.jackson.version>
```
Comment (mbasiuk-talend) [inline components-parent/pom.xml]: Is this conform to daikon's?
https://github.com/Talend/daikon/blob/master/daikon/pom.xml#L65-L84

File: components-parent/pom.xml [hunk 2]
```diff
@@ -1,634 +1,1562 @@
...
+        <commons-httpclient.version>3.1</commons-httpclient.version>
+        <commons-io.version>2.5</commons-io.version>
+        <commons-lang.version>2.6</commons-lang.version>
+        <commons-lang3.version>3.4</commons-lang3.version>
+        <commons-logging-api.version>1.1</commons-logging-api.version>
+        <commons-logging.version>1.2</commons-logging.version>
+        <commons-math.version>2.1</commons-math.version>
+        <commons-math3.version>3.4.1</commons-math3.version>
+        <commons-modeler.version>2.0.1</commons-modeler.version>
+        <commons-net.version>2.2</commons-net.version>
```
Comment (mbasiuk-talend) [inline components-parent/pom.xml]: com.marklogic:mlcp-9.0.3 needs: org.apache.hadoop:hadoop-common-2.6.0 -> commons-net:commons-net-3.1 (2.2 may break)

Comment (undx) [reply components-parent/pom.xml]: Setting to latest commons-net:commons-net-3.6

File: components-parent/pom.xml (hunk 1 above)
Comment (undx) [reply components-parent/pom.xml]: components are bounded to daikon, but it is another project managing its deps.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1637/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1637/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1638/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1638/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1639/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1639/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1640/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1640/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1641/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1641/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1642/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1642/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39964_CommonJarsAlignment)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1643/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1643/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39964_CommonJarsAlignment)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1648/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1648/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1649/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1649/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39964_CommonJarsAlignment)

**Comment by undx** [comment]
Don't know why `component-jdbc-runtime` sporadicaly fails... :unamused:
Just retry the build and it passes! :sunny:
**Failed**
`org.talend.components.jdbc.JDBCOutputTestIT.org.talend.components.jdbc.JDBCOutputTestIT`

Failing for the past 1 build (Since Failed#1648 )
Took 1 min 0 sec.

*Error Message*
A lock could not be obtained within the time requested
```Stacktrace
java.sql.SQLTransactionRollbackException: A lock could not be obtained within the time requested
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source)
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source)
	at org.apache.derby.impl.services.locks.ConcurrentLockSet.lockObject(Unknown Source)
	at org.apache.derby.impl.services.locks.AbstractPool.lockObject(Unknown Source)
	at org.apache.derby.impl.services.locks.ConcurrentPool.lockObject(Unknown Source)
	at org.apache.derby.impl.store.raw.xact.ContainerLocking3.lockContainer(Unknown Source)
	at org.apache.derby.impl.store.raw.data.BaseContainerHandle.useContainer(Unknown Source)
	at org.apache.derby.impl.store.raw.data.BaseDataFileFactory.openContainer(Unknown Source)
	at org.apache.derby.impl.store.raw.data.BaseDataFileFactory.openContainer(Unknown Source)
	at org.apache.derby.impl.store.raw.xact.Xact.openContainer(Unknown Source)
	at org.apache.derby.impl.store.access.conglomerate.OpenConglomerate.init(Unknown Source)
	at org.apache.derby.impl.store.access.heap.Heap.open(Unknown Source)
	at org.apache.derby.impl.store.access.RAMTransaction.openConglomerate(Unknown Source)
	at org.apache.derby.impl.store.access.RAMTransaction.openConglomerate(Unknown Source)
	at org.apache.derby.impl.sql.execute.DDLConstantAction.lockTableForDDL(Unknown Source)
	at org.apache.derby.impl.sql.execute.DropTableConstantAction.executeConstantAction(Unknown Source)
	at org.apache.derby.impl.sql.execute.MiscResultSet.open(Unknown Source)
	at org.apache.derby.impl.sql.GenericPreparedStatement.executeStmt(Unknown Source)
	at org.apache.derby.impl.sql.GenericPreparedStatement.execute(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedStatement.execute(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedStatement.execute(Unknown Source)
	at org.talend.components.jdbc.common.DBTestUtils.dropTestTable(DBTestUtils.java:103)
	at org.talend.components.jdbc.JDBCOutputTestIT.afterClass(JDBCOutputTestIT.java:63)```

**Comment by pteyssier** [comment]
Build and integration tests + unit tests ok

**Comment by undx** [comment]
Thanks all for the review and thanks Pierre for the merge 👍

**Pull request merged**